### PR TITLE
WARCSpout doesn't handle http.content.limit -1 correctly, fixes #850

### LINKED ISSUE #850: WARCSpout doesn't handle http.content.limit -1 correctly
-1 is the value it takes by default if not set in the config and it means that the content should not be trimmed, just like the Fetcher Bolts do, however it doesn't handle it correctly and tries to trim to a negative value.
A temporary workaround is to configure it with a large value

`  http.content.limit: 99999999`

### PR DESCRIPTION
**Pull request by sebastian-nagel**
- if http.content.limit is -1 trim content to the max. possible array size

The following issues where fixes to allow testing with 3 WARC files holding content of [2^31-8 (safe)](https://github.com/DigitalPebble/storm-crawler/files/5880994/test-size-int-max-safe.warc.gz), [2^31-1 (int_max)](https://github.com/DigitalPebble/storm-crawler/files/5880999/test-size-int-max.warc.gz) and [2^31 bytes](https://github.com/DigitalPebble/storm-crawler/files/5881003/test-size-int-max-overflow.warc.gz) and to fix bugs detected while testing these edge cases:
- when reading into multiple buffers: do not read more bytes than necessary
- avoid an integer overflow when reporting length of truncation
- implement gzip and deflate Content-Type decompression

### CODE REVIEW

File: external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCSpout.java [hunk 1]
```diff
@@ -59,6 +59,8 @@
 import com.digitalpebble.stormcrawler.spout.FileSpout;
 import com.digitalpebble.stormcrawler.util.ConfUtils;
 
+import parquet.it.unimi.dsi.fastutil.Arrays;
```
Comment (jnioche) [inline external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCSpout.java:62]: I presume this is already in our dependencies but I'd rather just have a static final somewhere in our code

Comment (kkrugler) [reply external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCSpout.java:62]: Is this a shaded class? It's `it.unimi.dsi.fastutil.Arrays` in my code. If it's shaded, you don't want to be using it in your code.

Comment (sebastian-nagel) [reply external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCSpout.java:62]: Ok, I'll change it. Also to be used for #854.

Comment (sebastian-nagel) [reply external/warc/src/main/java/com/digitalpebble/stormcrawler/warc/WARCSpout.java]: Ok, it's now defined in the class Constants:
```java
/** Maximum array size, safe value on any JVM */
public static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
```

**Comment by jnioche** [comment]
thanks @sebastian-nagel and @kkrugler
